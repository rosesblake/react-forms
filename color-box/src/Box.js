import React, { useState } from "react";

const Box = ({ color, width, height, deleteBox, box }) => {
  return (
    <div
      data-testid="box"
      className="Box"
      style={{
        backgroundColor: color,
        width: `${width}px`,
        height: `${height}px`,
      }}
    >
      <button onClick={() => deleteBox(box)}>X</button>
    </div>
  );
};

export default Box;
