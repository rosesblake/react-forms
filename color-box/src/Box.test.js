import React from "react";
import { render, fireEvent } from "@testing-library/react";
import Box from "./Box";

it("renders", () => {
  render(<Box />);
});

it("matches snapshot", () => {
  const { asFragment } = render(<Box color="blue" />);
  expect(asFragment()).toMatchSnapshot();
});
