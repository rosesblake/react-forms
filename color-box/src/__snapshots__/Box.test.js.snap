// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Box"
    data-testid="box"
    style="background-color: blue;"
  >
    <button>
      X
    </button>
  </div>
</DocumentFragment>
`;
